topcoffea/modules/histEFT.py

```diff
@@ -318,6 +317,26 @@ def __reduce__(self):
             ),
         )
 
+    def make_scalings(self,wc_scalings_lst=None):
+        scalingsbins = np.array(self.values(flow=True)[1:])[:, 1:-1]
+        scalingsbins_normalized = np.nan_to_num(scalingsbins / scalingsbins[:,0].reshape(-1,1), nan=0.0)
+        scalings = []
+        for numbers in scalingsbins_normalized:
+            n = len(self._wc_names) + 1
+            lower_matrix = np.zeros((n, n))
+            idx = 0
+
+            for i in range(n):
+                lower_matrix[i, :i + 1] = numbers[idx:idx + i + 1]
+                idx += i + 1
+
+            lower_matrix[np.tril_indices(n, -1)] /= 2
+            scalings.append(lower_matrix[np.tril_indices(n)].tolist())
+        if wc_scalings_lst and wc_scalings_lst != self._wc_names:
+            scalings = efth.remap_coeffs(self._wc_names,wc_scalings_lst,np.array(scalings)).tolist()
+
+        return scalings
```

Comment:
This entire method is in dire need of some comments/explanation. It would be nice to have a `docstring` outlining what the method _wants_ to do and what it expects. Then also some comments in the body as well explaining how it goes about accomplishing that.

This is particularly important for the lines that utilize array slicing, since things like numbers `0`, `1`, `2`, etc. are extremely non-descriptive in a lot cases.

Reply:
let's also try to find a better/more descriptive name than `numbers` for the iterator variable in the outermost `for` loop. I won't bore you with repeating my favorite joke in computer science here, even though it's (as always!) relevant.

Reply:
@Andrew42, 
Thank you for the comments, the function now has more detailed explanations. However, for the slicing, except for the underflow and overflow bins there are also extra coefficients of 0 at the beginning and the end of each bin, which we should discuss more of what those mean.